reciprocalspaceship/algorithms/intensity.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+import reciprocalspaceship as rs
+
+
+def compute_intensity_from_structurefactor(
+    ds,
+    F_key,
+    SigF_key,
+    output_columns=None,
+    dropna=True,
+    inplace=False,
+):
+    """
+    Back-calculate intensities and approximate intensity error estimates from
+    structure factor amplitudes and error estimates
+
+    Intensity computed as I = SigF*SigF + F*F. Intensity error estimate
+    approximated as SigI = abs(2*F*SigF)
+
+    Parameters
+    ----------
+    ds : DataSet
+        Input DataSet containing columns with F_key and SigF_key labels
+    F_key : str
+        Column label for structure factor amplitudes to use in back-calculation
+    SigF_key : str
+        Column label for structure factor error estimates to use in
+        back-calculation
+    output_columns : list or tuple of column names
+        Column labels to be added to ds for recording back-calculated I and
+        SigI, respectively. output_columns must have len=2.
+    dropna : bool
+        Whether to drop reflections with NaNs in F_key or SigF_key
+        columns
+    inplace : bool
+        Whether to modify the DataSet in place or create a copy
```

Comment:
I was originally thinking that this function would always return a new DataSet that just had [F, SigF,  I, SigI] as the columns. This would avoid the need for `output_columns` and `inplace` arguments. 

Do you think there is a significant advantage to letting this function be performed `inplace` on a given DataSet? If so, this can be modified to use the `@inplace` decorator to avoid needing to implement it yourself -- see functions in `dataset.py` for example usage.

Reply:
Ah,  you're right --  I had thought`rs.algorithms.scale_merged_intensities()` always returned a new DataSet. In that case, I think it's fine to do it this way for consistency. 

I still think the `@inplace` decorator should be used to provide that functionality -- the FW code predates that decorator, and it should be updated in the near future for consistency.